airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -359,7 +359,7 @@ def create_declarative_stream(model: DeclarativeStreamModel, config: Config, **k
     if model.schema_loader:
         schema_loader = _create_component_from_model(model=model.schema_loader, config=config)
     else:
-        schema_loader = DefaultSchemaLoader(config=config, options=model.options)
+        schema_loader = DefaultSchemaLoader(config=config, name=model.name, options=model.options)
```

Comment:
In the case of the connector-builder, it seems like the stream name is passed [in the request body](https://github.com/airbytehq/airbyte/blob/fcf193db16b65c25f56bf7650e6537658aafc1b8/airbyte-connector-builder-server/connector_builder/generated/models/stream_read_request_body.py#L27). Could we use this to populate the name downstream through the options? 

To be clear, I would prefer to have "name" if "name" is necessary for this class to do the job. Right now, it seems only for debugging purposes though

Reply:
The connector builder is actually something of a unique case. The main point is that connectors that use the connector builder UI don't actually support this type Schema Loader. It has to use an inline schema loader since the builder doesn't support multiple files.

But specifically for the `stream` name that is passed in via the request to test YAML that you pointed out, we don't want that to directly impact the YAML being tests. Think of that parameter as telling us which stream we want to test rather than something we inject into the manifest itself. So for example, we might have 4-5 streams in the YAML manifest. And the `stream` parameter of StreamReadRequestBody tells us to take the given manifest and test that stream.